core/dbt/parser/partial.py

```diff
@@ -417,6 +417,14 @@ def schedule_nodes_for_parsing(self, unique_ids):
                 self._schedule_for_parsing(
                     "metrics", metric, metric.name, self.delete_schema_metric
                 )
+            elif unique_id in self.saved_manifest.metrics:
```

Comment:
Shouldn't this be elif unique_id in self.saved_manifest.semantic_models: ?

Reply:
Yes.